diff --git a/packages/minio/0/templates/_helpers.tpl b/packages/minio/0/templates/_helpers.tpl
index 1cb209e..56aced1 100644
--- a/packages/minio/0/templates/_helpers.tpl
+++ b/packages/minio/0/templates/_helpers.tpl
@@ -132,7 +132,13 @@ Also, we can not use a single if because lazy evaluation is not an option
 {{- if .Values.global.imagePullSecrets }}
 imagePullSecrets:
 {{- range .Values.global.imagePullSecrets }}
-  - name: {{ . }}
+  {{- $imagePullSecrets := list }}
+  {{- if kindIs "string" . }}
+    {{- $imagePullSecrets = append $imagePullSecrets (dict "name" .) }}
+  {{- else }}
+    {{- $imagePullSecrets = append $imagePullSecrets . }}
+  {{- end }}
+  {{- toYaml $imagePullSecrets | nindent 2 }}
 {{- end }}
 {{- else if .Values.imagePullSecrets }}
 imagePullSecrets:
